### PR TITLE
Adds move score defines + minor score clean up

### PR DESCRIPTION
**Pull request by AlexOn1ine**
Only covers `AI_TryToFaint` and `AI_CalcMoveScore` for now. The can either be merged or I will keep going in the next few days.     

Minor note: If I removed stuff it means it is handled elsewhere, usually `AI_CalcDamage`

### CODE REVIEW

**Comment by Bassoonian** [comment]
You'll need to update `gAbilities` to `gAbilitiesInfo` for this to compile

**Comment by AlexOn1ine** [comment]
> You'll need to update `gAbilities` to `gAbilitiesInfo` for this to compile

Yeah, I messed up a bit. I'll fix it later when I do that rest.

**Comment by AlexOn1ine** [comment]
Ready for re-review.

I left out a lot of the big and minus score out. I'm not sure how to handle them for now. we can't uses negative defines, right? Maybe a small macro would be fine here.

**Comment by Bassoonian** [comment]
Still failing to build

**Comment by AlexOn1ine** [comment]
> Still failing to build

agbcc complained. Fixed

File: include/battle_ai_main.h [hunk 1]
```diff
@@ -29,6 +29,25 @@
 #define STAT_CHANGE_ACC        10
 #define STAT_CHANGE_EVASION    11
 
+#define BEST_DAMAGE_MOVE 1 // Move with the most amount of hits with the best accuracy/effect
+
+// Temprorary scores that are added together to determine a final score at the at of AI_CalcMoveScore
```
Comment (Bassoonian) [inline include/battle_ai_main.h:34]:
```suggestion
// Temporary scores that are added together to determine a final score at the at of AI_CalcMoveScore
```

**Review by Bassoonian — APPROVED** [review]

**Pull request merged**